ios/ReactNativePageView.m

```diff
@@ -10,412 +9,333 @@
 
 @interface ReactNativePageView () <UIPageViewControllerDataSource, UIPageViewControllerDelegate, UIScrollViewDelegate>
 
-@property(nonatomic, strong) UIPageViewController *reactPageViewController;
+@property(nonatomic, strong) NSMapTable<UIView *, UIViewController *> *controllerCache;
+@property NSInteger currentPage;
+@property NSNumber *currentReactTag;
+@property(nonatomic, strong) NSMapTable<UIViewController *, NSNumber *> *placeholderPageIndexes;
 @property(nonatomic, strong) UIPageControl *reactPageIndicatorView;
-@property(nonatomic, strong) RCTEventDispatcher *eventDispatcher;
-
+@property(nonatomic, strong) UIPageViewController *reactPageViewController;
 @property(nonatomic, weak) UIScrollView *scrollView;
-@property(nonatomic, weak) UIView *currentView;
-
-@property(nonatomic, strong) NSHashTable<UIViewController *> *cachedControllers;
-@property (nonatomic, assign) CGPoint lastContentOffset;
-
-- (void)goTo:(NSInteger)index animated:(BOOL)animated;
-- (void)shouldScroll:(BOOL)scrollEnabled;
-- (void)shouldShowPageIndicator:(BOOL)showPageIndicator;
-- (void)shouldDismissKeyboard:(NSString *)dismissKeyboard;
-
 
 @end
 
-@implementation ReactNativePageView {
-    uint16_t _coalescingKey;
-}
 
-- (instancetype)initWithEventDispatcher:(RCTEventDispatcher *)eventDispatcher {
-    if (self = [super init]) {
-        _scrollEnabled = YES;
-        _pageMargin = 0;
-        _previousIndex = -1;
-        _transitionStyle = UIPageViewControllerTransitionStyleScroll;
-        _orientation = UIPageViewControllerNavigationOrientationHorizontal;
-        _currentIndex = 0;
-        _dismissKeyboard = UIScrollViewKeyboardDismissModeNone;
-        _coalescingKey = 0;
-        _eventDispatcher = eventDispatcher;
-        _cachedControllers = [NSHashTable weakObjectsHashTable];
-        _overdrag = YES;
-    }
-    return self;
-}
+@implementation ReactNativePageView
 
-- (void)layoutSubviews {
-    [super layoutSubviews];
-    if (self.reactPageViewController) {
-        [self shouldScroll:self.scrollEnabled];
-        //Below line fix bug, where the view does not update after orientation changed.
-        [self updateDataSource];
-    }
-}
-
-- (void)didUpdateReactSubviews {
-    if (!self.reactPageViewController && self.reactViewController != nil) {
+- (instancetype)init {
+    if (self = [super init]) {
+        self.controllerCache = [NSMapTable weakToWeakObjectsMapTable];
```

Comment:
Why self instead of _ ?

Reply:
`_` and `self` seemed to be used interchangeably, without any obvious reason for one over the other.  I choose to use just one, for consistency.

Looked it up now, and appears `_` is more efficient for basic assignment, since it bypasses setters.  I can switch to that.  However, I am quite unfamiliar with Objective C.  Are there places other than the constructor that should also be switched?

Reply:
> `_` and `self` seemed to be used interchangeably, without any obvious reason for one over the other. I choose to use just one, for consistency.
> 
> Looked it up now, and appears `_` is more efficient for basic assignment, since it bypasses setters. I can switch to that. However, I am quite unfamiliar with Objective C. Are there places other than the constructor that should also be switched?

I am not familiar with objC too, but I would prefer to use `_` in each place instead of `self`. Could you revert it to _ ? 

> Are there places other than the constructor that should also be switched?

I think, the whole file should contain `_` to keep consistency